### ISSUE TITLE
admin comments

### ISSUE THREAD

**Issue by Th14go** [opensupports/opensupports#287]
Sorry for the use of the translator.
I hope I can pass the error, I did not find anything in the issues, related to the error that I am facing, the user creates the ticket, in the admin panel, his ticket appears, I answer him, however in the user this answer does not appear, only when he closes the session and login again. What can it be ??

[Picture](https://imgur.com/GRbj7p3)

no reply appears while logout and login again..

I also noticed that when creating a new ticket that updates there and shows a response of the previous tickets

Thank you for making this project public right now.

att: Thiago Lopes

**Comment by carlopilurzi** [comment]
first of all thanks for the work vs, I have the same problem, the response window covers the response of the staff, if the user log out and then log in, at this point can see the answer. If it is useful to use chrome to navigate.
See you soon,
Carlo

**Comment by mredigonda** [comment]
This is the first time we hear from this issue, we will be happy to hear more details:
1. Does it happen using Chrome? Firefox? both? (please try with the most recent versions of those)
2. What is the output from the console (F12 on Chrome)?
Thank you for reporting!

**Comment by Th14go** [comment]
mredigonda , This is not the current, and the are existing in the latest version, you can that make a screen-record?

**Comment by carlopilurzi** [comment]
Hi Maximiliano, I use chrome 68.0.3440.106 (Official Build) (64-bit) on Mac, and the problem occurs if the user writes the request and waits for the response without logging out.
Staff response is not visible unless you log out and log in.
It would be very useful if the user page occasionally went to check if there is a response from the staff and make it visible.
sorry for the automatic translator and thanks for your excellent work vs.
Carlo
I enclose the screen after the team response without log out of the user and after the log out and subsequent user log in.
<img width="1200" alt="1" src="https://user-images.githubusercontent.com/42648941/45545184-dd391900-b819-11e8-9fff-80ea4d83c580.png">
<img width="1186" alt="2" src="https://user-images.githubusercontent.com/42648941/45545186-e1fdcd00-b819-11e8-8e3b-25983889455a.png">

**Comment by nickbe** [comment]
Very good point. I'm not sure if this is a bug or just not implemented. Same problem when you are logged in as staff. Somehow there's no sure way to see when a user responds or anyone writes a new ticket except pressing F5 from time to time.

@mredigonda Could this enhaced somehow?

**Comment by Th14go** [comment]
It is exactly what is happening through the print of @carlopilurzi.

**Comment by nickbe** [comment]
What do you mean by "print" ?

**Comment by Th14go** [comment]
"screenshot"....

**Comment by carlopilurzi** [comment]
Good morning, I wanted to add that, for the staff, if you update the page you can see the new comments added by the user, while the problem is only for users who also updating the page, can not see the comments of the staff, we must do logout and a new login to see the comment added.
thank you
Carlo

**Comment by Th14go** [comment]
In my case, neither with F5, appear the comments, only with logout and login.

**Comment by nickbe** [comment]
I think this is a major problem right now. Even F5 should not be necessary when checking for new tickets and/or responses of any kind.

Could our developers look into this please? 😄

**Comment by ivandiazwm** [comment]
Absolutely. Tickets updates should not stay this way because it looks confusing to users.

I have a couple of suggestions:
1) Refreshing the tickets/comments every time the list is rendered. It'll work like F5 (because currently if you move to a tab and return to the ticket list, it doesn't update).
2) Refreshing the tickets/comments every `n` seconds with some sort of short/long polling.
3) Adding a "refresh" button, so we avoid unnecessary requests.

I'm thinking that a good solution may be to have automated refreshing for users and a refresh button for staffs members.

What do you think? Any other suggestions?

**Comment by nickbe** [comment]
In my opinion you should choose the easiest way possible for users and staff. A general automatic polling delay of n seconds is definetely the best.  Being logged in as staff would usually mean you stay online, waiting in the background for new tickets to arrive. Pressing anything manually ... personally I would simply forget to.

**Comment by Th14go** [comment]
If you post an automatic update every 5 seconds, depending on no user action or team action. ?

**Comment by Th14go** [comment]
If you post an automatic update every 5 seconds, depending on no user action or team action. ?

**Comment by Th14go** [comment]
If you enter an automatic update every 5 seconds, not depending on any user action or team action. ?

**Comment by tarasian666** [comment]
> Absolutely. Tickets updates should not stay this way because it looks confusing to users.
> 
> I have a couple of suggestions:
> 
> 1. Refreshing the tickets/comments every time the list is rendered. It'll work like F5 (because currently if you move to a tab and return to the ticket list, it doesn't update).
> 2. Refreshing the tickets/comments every `n` seconds with some sort of short/long polling.
> 3. Adding a "refresh" button, so we avoid unnecessary requests.
> 
> I'm thinking that a good solution may be to have automated refreshing for users and a refresh button for staffs members.
> 
> What do you think? Any other suggestions?

implement at least Your first suggestion

**Comment by nickbe** [comment]
Nice. Thanks a lot.

**Comment by tarasian666** [comment]
> Absolutely. Tickets updates should not stay this way because it looks confusing to users.
> 
> I have a couple of suggestions:
> 
> 1. Refreshing the tickets/comments every time the list is rendered. It'll work like F5 (because currently if you move to a tab and return to the ticket list, it doesn't update).
> 2. Refreshing the tickets/comments every `n` seconds with some sort of short/long polling.
> 3. Adding a "refresh" button, so we avoid unnecessary requests.
> 
> I'm thinking that a good solution may be to have automated refreshing for users and a refresh button for staffs members.
> 
> What do you think? Any other suggestions?

Also, there will be another issue, Your system request all data at the same time (all tickets and comments etc), this can cause long loading, 504 error etc, my suggestion - request only tickets list and last comments in dashboard, full ticket conversation request only on "ticket opened" event.

**Comment by nickbe** [comment]
This would explain why our test system behaved a little strange after adding many test tickets. I thought this was because there's some index missing in the database definition. But if all tickets are being loaded all the time then this is surely bound to cause trouble.

@ivandiazwm - Can this be changed without too much technical trouble?

**Comment by tarasian666** [comment]
temporary hack
```
    initSession() {
        return {
            type: 'CHECK_SESSION',
            payload: new Promise((resolve, reject) => {
                API.call({
                    path: '/user/check-session',
                    data: {}
                }).then((result) => {
                    if (!result.data.sessionActive) {
                        if (sessionStore.isRememberDataExpired()) {
                            store.dispatch({
                                type: 'LOGOUT_FULFILLED'
                            });
                        } else {
                            store.dispatch(this.autoLogin());
                        }

                        resolve(result);
                    } else if(sessionStore.isLoggedIn()) {
                        let sessionData = sessionStore.getSessionData();
                        let userData = sessionStore.getUserData();
                        store.dispatch(this.getUserData(sessionData.userId, sessionData.token, userData.staff));
                        store.dispatch({
                            type: 'SESSION_CHECKED'
                        });

                        resolve(result);
                    } else {
                        reject(result);
                    }
                });
            })
        }
    }
```

**Comment by zlobik4** [comment]
> temporary hack, updates tickets on F5

Could you tell me please where to insert/put it?
Which path and file?

**Comment by tarasian666** [comment]
> > temporary hack, updates tickets on F5
> 
> Could you tell me please where to insert/put it?
> Which path and file?

session-actions.js just replace all initSession() function or look for the difference

**Comment by ivandiazwm** [comment]
I've changed this logic and now it retrieves the tickets every time the list is rendered. So it will work with F5.

This fix will be available for next release `v4.4`.

**Comment by Th14go** [comment]
Good morning, @ivandiazwm, you are scheduled to release this version 4.4 when ?
